src/FriendPanel.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useRef } from 'react';
+import { flushSync } from 'react-dom';
+import './FriendPanel.css';
+
+export default function FriendPanel({selectFriend, selectedFriend, message, setMessage, history, setHistory}) {
+    const [friends, setFriends] = useState([]);
+    const [newFriend, setNewFriend] = useState('');
+    const selectedRef = useRef(null);
+  
+    const handleChangeFriend = (event) => {
+      setNewFriend(event.target.value);
+    };
+  
+    function addFriend() {
+      if (friends.includes(newFriend)) {
+        flushSync(() => {selectFriend(newFriend)});
+        selectedRef.current.scrollIntoView();
```

Comment:
Instead of using ref to scroll into correct element, you can add `id` attribute to relevant elements and scroll into that using sth like this:
```
const element = document.getElementById('<unique-friend-id>');
element.scrollIntoView();
```
And you would probably add these ids to `button` in L63:
```
<button ... id={'<unique-friend-id>'} >...</button>
```